packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -509,8 +512,11 @@ export function NativeStackView({
           // On Fabric, when screen is frozen, animated and reanimated values are not updated
           // due to component being unmounted. To avoid this, we don't freeze the previous screen there
           const shouldFreeze = isFabric()
-            ? !isPreloaded && !isFocused && !isBelowFocused
-            : !isPreloaded && !isFocused;
+            ? !isPreloaded &&
+              !isFocused &&
+              !isBelowFocused &&
+              !(Platform.OS === 'ios' && isModal)
+            : !isPreloaded && !isFocused && !(Platform.OS === 'ios' && isModal);
```

Comment:
nit, but this check is already hard :D 
```suggestion
const isModalOnIos = (Platform.OS === 'ios' && isModal)

            ? !isPreloaded &&
              !isFocused &&
              !isBelowFocused &&
              !isModalOnIos
            : !isPreloaded && !isFocused && !isModalOnIos;
```